plugins/query-insights/src/main/java/org/opensearch/plugin/insights/rules/model/Measurement.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.rules.model;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Represents a measurement with a name and a corresponding value.
+ *
+ * @param <T> the type of the value
+ */
+public class Measurement<T extends Number & Comparable<T>> {
+    String name;
+    T value;
+
+    /**
+     * Create a Measurement from a StreamInput
+     * @param in the StreamInput to read from
+     * @throws IOException IOException
+     */
+    @SuppressWarnings("unchecked")
+    public Measurement(StreamInput in) throws IOException {
+        this.name = in.readString();
+        this.value = (T) in.readGenericValue();
+    }
+
+    /**
+     * Write Measurement to a StreamOutput
+     * @param out the StreamOutput to write
+     * @param measurement the Measurement to write
+     * @throws IOException IOException
+     */
+    public static void writeTo(StreamOutput out, Measurement<?> measurement) throws IOException {
+        out.writeString(measurement.getName());
+        out.writeGenericValue(measurement.getValue());
+    }
+
+    /**
+     * Constructs a new Measurement with input name and value.
+     *
+     * @param name  the name of the measurement
+     * @param value the value of the measurement
+     */
+    public Measurement(String name, T value) {
+        this.name = name;
+        this.value = value;
+    }
```

Comment:
What are the possible values of name for measurement class? I am wondering if we really need the Measurement class

Reply:
The possible values are defined in the MetricType. Measurement class is needed to make the record class more generic. This was suggested and discussed with .rishabhmaurya in this comment: https://github.com/opensearch-project/OpenSearch/pull/11903#discussion_r1466974170

Reply:
Is it not possible to use just the `? extends Number` wherever we are using `Measurement<? extends Number>`? Also, serialization might look weird `"JVM" : {"JVM": 5.0}` with the current model

Reply:
And, original comment is about condensing the SearchQueryRecord class, which I think is good idea

Reply:
We need the stream support for reading/writing a Measurement value, and the general `Number` Class won't provide us those functionalities. So we need a dedicated class `Measurement` to handle the stream operations. Also, the general `Number` class is not `Comparable` neither.

But let me see if we can simplify the logic here.

Reply:
also, can you avoid such repetitive keys `"JVM" : {"JVM": 5.0}` and still have measurement class.

Reply:
Resolved this with @jainankitk and refactored the code.